package_test.go

```diff
@@ -669,8 +673,9 @@ func (s *PackageSuite) TestQueryMultipleRuns(c *C) {
 		if i >= len(iterOutputs) {
 			c.Fatalf("expected %d rows, got more", len(iterOutputs))
 		}
-		if !iter.Decode(iterOutputs[i]) {
-			break
+		if err := iter.Decode(iterOutputs[i]); err != nil {
+			iter.Close()
+			c.Fatal(err)
```

Comment:
The `iter.Close` here seems a bit arbitrary, in the sense that there's a different fatal call a couple of lines above and the iterator is not being closed there. I'd suggest instead doing a `defer iter.Close()` at the top.

Also, perhaps not for this PR, but all of these Fatal calls can be changed to:
```
err := .... do whatever ...
c.Assert(err, IsNil)
```